webapp/public/run.html

```diff
@@ -168,8 +169,14 @@
                     window.parent.postMessage(msg.data, "*");
                 }
                 else if (mpRole && data.type === "multiplayer") {
-                    // propagate message packets to parent frames
-                    window.parent.postMessage(msg.data, "*");
+                    // Multiplayer messages are bi-directional
+                    if (mpRole === msg.data.origin) {
+                        // propagate to parent frame
+                        window.parent.postMessage(msg.data, "*");
+                    } else {
+                        // propagate to child frames
+                        if (simdriver) simdriver.postMessage(msg.data);
```

Comment:
I'm not certain I understand the need for posting to simdriver here -- as those aren't getting lost / get handled elsewhere right as it had previously gotten sent to the sim iframe, so would this double-send the message?

Reply:
The intention here is to simply forward multiplayer messages along. Messages from the parent are forwarded to the simulator (via simdriver). Messages from the simulator are forwarded to the parent. It isn't clear how a double-send could happen. Is there something I'm not considering? There are so many message handlers scattered about. I might have missed where these messages are forwarded.

Reply:
ah, what I meant was that if it was working on localhost, then the messages were getting passed to child frames somewhere else -- which would mean adding this would add a second path that intended to pass the same message from run.html -> simframes, right (with the first path not working right currently on live which might be related to missing parentOrigin)?